### DISCUSSION TITLE
How many cores cloudprober need to detect 10K server when average response time is about 100us?

### DISCUSSION THREAD

**Discussion by kamushin** [cloudprober/cloudprober#164] [Q&A]
How many cores cloudprober need to detect 10K servers when average response time is about 100us?

**Comment by manugarg** [comment]
Do you mean for probing 10K servers? If it's PING probe, not much at all. For HTTP probe, it depends.

For HTTP probe, if you're using `keep_alive: true`, it should not be very costly, as establishing a new TCP connection is where it starts to get more expensive with more targets. Another cost will be around multiplexing between 10k goroutines as Cloudprober run one goroutine per target for HTTP probing. But, even then, I don't think it will not be super expensive.

Have you tried at all?

**Comment by syd20** [comment]
@kamushin If it helps, I have been able to test ~9000 http probes on a instance with 1 core and 2GB RAM
